src/components/modal/Modal.jsx

```diff
@@ -0,0 +1,42 @@
+import React from 'react';
+import './Modal.css';
+
+const Modal = ({
```

Comment:
API wise, you are missing the two functionalities I asked for: 

**1**. Allowing a different component to be the close button. With this API I cannot use a link instead of a button, or simply change the contents of the button (to render "close" instead of "x".

**2**. Being a controlled component it's possible to close the modal from within the content. The state is outside the component and you just set that given state. What I want is a Modal that can work either as a controlled component (the normal mode) or as a uncontrolled one. 

```jsx
<Modal isClosed={false} />
<Modal defaultIsClosed={true} onClose={() =>}/>
```

Being uncontrolled you'll need a specific API to allow closing the modal from a children. You can do that in two ways that I can think of.  Also notice that if you can pass the initial state, you may also need an API to open the modal imperatively.

Give it a thought. You can run your ideas by me before making changes.